schematic/store/synapse.py

```diff
@@ -2188,70 +2267,73 @@ def _update_table_uuid_column(synStore: SynapseStorage, table_id: str, sg: Schem
                 if uuid_col_in_schema:
                     new_col = Column(columnType = "STRING", maximumSize = 64, name = "Id")
                     schema.addColumn(new_col)
-                    schema = synStore.syn.store(schema)
+                    schema = self.synStore.syn.store(schema)
                 # If there is not, then use the old `Uuid` column as a basis for the new `Id` column
                 else:
-                    # Create a new `Id` column based off of the old `Uuid` column, and store (column is empty)
-                    new_col = deepcopy(col)
-                    new_col['name'] = 'Id'
-                    schema.addColumn(new_col)
-                    schema = synStore.syn.store(schema)
-                
-                
-                    # Recently stored column is empty, so populated with uuid values
-                    TableOperations._populate_new_id_column(synStore, table_id, schema)
-
-                    # get the up-to-date table, remove old `Uuid` column, and store
-                    sleep(1)
-                    schema = synStore.syn.get(table_id)
-                    schema.removeColumn(col)
-                    schema = synStore.syn.store(schema)
+                    
+                    # Build ColumnModel that will be used for new column
+                    columnModelDict = {
+                        "id": None,
+                        "name": "Id",
+                        "defaultValue": None,
+                        "columnType": "STRING",
+                        "maximumSize": 64,
+                        "maximumListLength": 1,
+                    }
+
+                    # Send POST /column request to define new column and get new column ID
+                    newColResponse = self.synStore.send_api_request(
+                        request_type = "restPOST",
+                        uri = "https://repo-prod.prod.sagebase.org/repo/v1/column",
+                        body = columnModelDict,
+                    )
```

Comment:
This can be done with the client:

```
new_col = synapseclient.Column(name='Id', columnType='STRING', maximumSize=64, maximumListLength=1)
self.synStore.syn.store(new_col)
```

Reply:
Would the value returned by `self.synStore.syn.store(new_col)` be the same as what we're currently getting now / does the `syn.store` operation actually store anything in this case or just create a column type object? Also, do you know if there's a performance difference between using the API and the client for this operation?

Reply:
Yes, the value would be the same.  `syn.store` just creates a new column type object, basically the only difference here is you don't need to test all of this since it's tested in the client.

Behind the scenes, the client just does the API call.  If you're interested, `syn.store` will look for this function in any class: https://github.com/Sage-Bionetworks/synapsePythonClient/blob/f3698af6c7d361c5a22d102d2625152933809ceb/synapseclient/table.py#L1517.  And it will call the `restPOST` command.  

You are essentially duplicating code and also formatting your own json when there's a Python object that you can use.

Reply:
Ahh I see, that makes sense. I think in the future we'll switch to using the client for this operation and the api call for the renaming

Reply:
@thomasyu888 for the example you gave, should the code be something like: 
```
from synapseclient.table import Column
new_col = Column(name='Id', columnType='STRING', maximumSize=64, maximumListLength=1) 
newColResponse = syn.store(new_col)
```

@GiaJordan if using the client directly is simplier, could you make the change in this PR and switch back to use the client? That will be easier to maintain in the future.

Reply:
I also dig into the client code more, and just like @thomasyu888 described, I think here is a duplicate of what the client is doing here: https://github.com/Sage-Bionetworks/synapsePythonClient/blob/f3698af6c7d361c5a22d102d2625152933809ceb/synapseclient/client.py#L1035

When storing a column, `restPOST` is taking: `obj.postURI()` which is `/column`, and `obj.json` is the json object: 
```
{ "columnType": "STRING", "concreteType": "org.sagebionetworks.repo.model.table.ColumnModel", "id": "192889", "maximumListLength": 1, "maximumSize": 64, "name": "Id" }
```
which is like what you are doing with the `send_api_request` function

Reply:
Yup, that's right!